### PR TITLE
Timer function

### PR DESCRIPTION
**Pull request by croustibaie**
Added a timer function. The timer is added in the "Task structure". 

While creating a new task, add a "task_duration" variable to the json.

### CODE REVIEW

File: worker/child/child.go [hunk 1]
```diff
@@ -54,7 +53,19 @@ func (s *Worker) Start() error {
...
+	resp, err:= s.t.c.GetTask(rpc.RequestTaskPayload{
+                TaskID: s.t.taskid,
+        })
+	if resp.TaskDuration !=0{ //If the task duration is 0 (not set), we don't run the timer
+		timer := time.NewTimer(time.Second * time.Duration(resp.TaskDuration))
+		go func() {
+			<-timer.C
+			log.Warn().Msg("Timer expired, stopping task")
+			s.t.Stop()
+			}()
```
Comment (tankbusta) [inline worker/child/child.go]: The indentation is off here

File: worker/child/child.go [hunk 2]
```diff
@@ -54,7 +53,19 @@ func (s *Worker) Start() error {
 		}
 	}()
 
-	s.t = NewTask(s.taskid, s.devices, s.cfg, s.rc)
+	s.t = NewTask(s.taskid, s.devices, s.cfg, s.rc) //Get the task in order to collect the task duration
+	resp, err:= s.t.c.GetTask(rpc.RequestTaskPayload{
+                TaskID: s.t.taskid,
+        })
+	if resp.TaskDuration !=0{ //If the task duration is 0 (not set), we don't run the timer
```
Comment (tankbusta) [inline worker/child/child.go]: Can you add a space after `!=`?

Comment (tankbusta) [reply worker/child/child.go]: Oh and a space after err, `resp, err := s.t.c.GetTask(...`

**Comment by tankbusta** [comment]
@croustibaie Do you want me to add this new field to the [UI component](https://github.com/fireeye/gocrack-ui)? I should be able to add it pretty quickly so we can get this out the door :)

File: worker/child/child.go (hunk 1 above)
Comment (croustibaie) [reply worker/child/child.go]: I'll correct this as soon as possible.

**Comment by croustibaie** [comment]
It would be awesome if you could add this to the UI! I'm actually far from being a dev and I have very few knowledge of node.js.

**Comment by tankbusta** [comment]
Awesome, this looks good. I'll get the UI portion done today, do some testing, and merge it!

**Comment by tankbusta** [comment]
Alright, I was able to create a component for the time limit [code here](https://github.com/fireeye/gocrack-ui/blob/feat/ui_timeout/src/components/CreateTask/TimeLimitInput.vue). 

It took a little longer than expected as I noticed a bunch of things were broken when I upgraded to the release build of Bootstrap 4, so I had to fix all those issues too. I should be able to test this and the LDAP fix you reported tomorrow

**Comment by croustibaie** [comment]
Great! Thanks a lot. Tell me if there is anything else I can do. 

I noticed Yesterday that connecting with a mix of upper and lower case letters while using the LDAP connector creates new accounts. Maybe lower casing the username in the LDAP authentication module could be nice. Still, some people could be using a case sensitive LDAP, so I am not really sure whether this is an issue or not.

**Comment by tankbusta** [comment]
Thanks for the contribution! 

I'll ask around if case-sensitive LDAP is a thing otherwise we can look at `strings.ToLower()`. It looks like it's very rare according to https://stackoverflow.com/questions/29897684/is-ldap-dn-case-insensitive

**Comment by tankbusta** [comment]
The UI portion was merged in if you want to pull it down and give it a try in your deployment :)

**Comment by croustibaie** [comment]
Hi, Thanks a lot, i'm testing it right away.

**Pull request merged**